### PR TITLE
Usage of long and unsigned long changed to platform independent fixed-sized types where appropriate.

### PR DESCRIPTION
**Pull request by rh101**

### CODE REVIEW

File: tests/cpp-tests/Classes/Sprite3DTest/Sprite3DTest.cpp [hunk 1]
```diff
@@ -893,7 +893,7 @@ void AsyncLoadSprite3DTest::menuCallback_asyncLoadSprite(Ref* sender)
 
 void AsyncLoadSprite3DTest::asyncLoad_Callback(Sprite3D* sprite, void* param)
 {
-    auto index = (int32_t)param;
+    auto index = (long)param;
```
Comment (halx99) [inline tests/cpp-tests/Classes/Sprite3DTest/Sprite3DTest.cpp:896]: `intptr_t` or `uintptr_t` is the best pointer conversion integer type

Comment (halx99) [reply tests/cpp-tests/Classes/Sprite3DTest/Sprite3DTest.cpp:896]: Maybe `static_cast<int>((uintptr_t)param))` is better?

Comment (rh101) [reply tests/cpp-tests/Classes/Sprite3DTest/Sprite3DTest.cpp:896]: Actually yes, that's a good point. I'll change it.

**Comment by halx99** [comment]
All `%ld` needs change to `%d` for int32_t or int

File: cocos/base/CCProfiling.cpp [hunk 2]
```diff
@@ -115,7 +115,7 @@ std::string ProfilingTimer::getDescription() const
 {
     static char s_description[512] = {0};
 
-    sprintf(s_description, "%s ::\tavg1: %ldu,\tavg2: %ldu,\tmin: %ldu,\tmax: %ldu,\ttotal: %.2fs,\tnr calls: %ld", _nameStr.c_str(), _averageTime1, _averageTime2, minTime, maxTime, totalTime/1000000., numberOfCalls);
+    sprintf(s_description, "%s ::\tavg1: %du,\tavg2: %du,\tmin: %du,\tmax: %du,\ttotal: %.2fs,\tnr calls: %d", _nameStr.c_str(), _averageTime1, _averageTime2, minTime, maxTime, totalTime/1000000., numberOfCalls);
```
Comment (halx99) [inline cocos/base/CCProfiling.cpp:118]: Shorten `%u` is better?

File: cocos/base/CCProperties.h [hunk 3]
```diff
@@ -337,7 +337,7 @@ class CC_DLL Properties
      * @return The value of the given property interpreted as a long.
      *   Zero if the property does not exist or could not be scanned.
      */
-    long getLong(const char* name = NULL) const;
+    int32_t getLong(const char* name = NULL) const;
```
[the remark this answers is not in the document]
Comment (rh101) [reply cocos/base/CCProperties.h:340]: There is a getInt already there.  I think this code is a little outdated, and should probably be changed to getInt32 and getInt64. Would that be better?

File: cocos/base/CCProfiling.cpp (hunk 2 above)
Comment (rh101) [reply cocos/base/CCProfiling.cpp:118]: Will do

File: cocos/base/TGAlib.h [hunk 4]
```diff
@@ -28,6 +28,8 @@ THE SOFTWARE.
 #define __SUPPORT_DATA_SUPPORT_TGALIB_H__
 /// @cond DO_NOT_SHOW
 
+#include <cstdint>
```
[the remark this answers is not in the document]
Comment (rh101) [reply cocos/base/TGAlib.h:31]: No worries, I'll change it

File: cocos/platform/CCPosixFileStream.cpp [hunk 5]
```diff
@@ -178,7 +178,7 @@ int PosixFileStream::close()
     return internalClose();
 }
 
-long PosixFileStream::seek(int64_t offset, int origin)
+int PosixFileStream::seek(int64_t offset, int origin)
 {
     const auto result = _iof->seek(_handle, static_cast<int32_t>(offset), origin); // this returns -1 for error, and resulting offset on success
```
Comment (halx99) [inline cocos/platform/CCPosixFileStream.cpp:183]: Needs explicit int64_t offset, and there are lseeki64 for windows and unix is int64_t by default

Comment (rh101) [reply cocos/platform/CCPosixFileStream.cpp:183]: Ok, I'll change it soon

File: cocos/base/CCProperties.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (halx99) [reply cocos/base/CCProperties.h:340]: I preferred remove getLong

File: cocos/platform/CCPosixFileStream.cpp (hunk 5 above)
Comment (halx99) [reply cocos/platform/CCPosixFileStream.cpp:183]: You can add
#if defined(_WIN32)
// ...
#  define posix_lseek64 ::_lseeki64
# else
#if defined(__APPLE__)
#  define posix_lseek64 ::lseek
#else
#  define posix_lseek64 ::lseek64
#endif
#endif

File: cocos/renderer/CCTexture2D.cpp [hunk 6]
```diff
@@ -284,7 +284,7 @@ bool Texture2D::updateWithMipmaps(MipmapInfo* mipmaps, int mipmapsNum, backend::
     auto& pfd = backend::PixelFormatUtils::getFormatDescriptor(pixelFormat);
     if (!pfd.bpp)
     {
-        CCLOG("cocos2d: WARNING: unsupported pixelformat: %lx", (unsigned long)pixelFormat);
+        CCLOG("cocos2d: WARNING: unsupported pixelformat: %lx", (uint32_t)pixelFormat);
```
Comment (halx99) [inline cocos/renderer/CCTexture2D.cpp:287]: Should be `%x`

File: cocos/base/CCProperties.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (rh101) [reply cocos/base/CCProperties.h:340]: No worries, I'll remove it

File: cocos/platform/CCPosixFileStream.cpp (hunk 5 above)
Comment (rh101) [reply cocos/platform/CCPosixFileStream.cpp:183]: Where should that section be added?

Comment (halx99) [reply cocos/platform/CCPosixFileStream.cpp:183]: https://github.com/adxeproject/adxe/blob/dev/cocos/platform/CCPosixFileStream.h

Comment (rh101) [reply cocos/platform/CCPosixFileStream.cpp:183]: All good, I'll sort it out soon

File: cocos/platform/CCPosixFileStream.cpp [hunk 7]
```diff
@@ -41,7 +41,7 @@ static int pfs_posix_open(const std::string& path, FileStream::Mode mode, PXFile
 
 // posix standard wrappers
 static int pfs_posix_read(PXFileHandle& handle, void* buf, unsigned int size) { return static_cast<int>(posix_read(handle._fd, buf, size)); }
-static off_t pfs_posix_seek(PXFileHandle& handle, off_t offst, int origin) { return posix_lseek(handle._fd, offst, origin); }
```
Comment (halx99) [inline cocos/platform/CCPosixFileStream.cpp]: Do you check the off_t on windows, I see the sizeof(off_t) on windows is 32bit

Comment (rh101) [reply cocos/platform/CCPosixFileStream.cpp]:
```
#ifndef __off_t_defined
# ifndef __USE_FILE_OFFSET64
typedef __off_t off_t;
# else
typedef __off64_t off_t;
# endif
# define __off_t_defined
#endif 
```

Do we want to define `__USE_FILE_OFFSET64`?

Comment (rh101) [reply cocos/platform/CCPosixFileStream.cpp]: Ok, do you want me to explicitly set it to int64_t instead of off_t?

Comment (halx99) [reply cocos/platform/CCPosixFileStream.cpp]: Yes

File: cocos/renderer/CCTexture2D.cpp (hunk 6 above)
Comment (rh101) [reply cocos/renderer/CCTexture2D.cpp]: Yep updated

**Comment by halx99** [comment]
Now, everything looks good to me

**Comment by rh101** [comment]
Excellent!

File: cocos/renderer/CCTextureCache.cpp [hunk 8]
```diff
@@ -652,14 +652,14 @@ std::string TextureCache::getCachedTextureInfo() const
         auto bytes = tex->getPixelsWide() * tex->getPixelsHigh() * bpp / 8;
         totalBytes += bytes;
         count++;
-        snprintf(buftmp, sizeof(buftmp) - 1, "\"%s\" rc=%lu id=%p %lu x %lu @ %ld bpp => %lu KB\n",
+        snprintf(buftmp, sizeof(buftmp) - 1, "\"%s\" rc=%lu id=%p %lu x %lu @ %d bpp => %lu KB\n",
```
[the remark this answers is not in the document]
Comment (rh101) [reply cocos/renderer/CCTextureCache.cpp]: done

**Pull request merged**